main.cu

```diff
@@ -95,6 +95,24 @@ __global__ void rank1_update_kernel(float* d_cov_matrix, const float* d_new_row,
     d_mean[idx] = new_mean;
 }
 
+__global__ void rank1_update_kernel_threaded(float* d_cov_matrix, const float* d_new_row, float* d_mean, const int cols, const int n) {
```

Comment:
this would be launched with a higher number of blocks/threads but again not much to gain given this would need to happen serially. different launch params with increased blocks/threads only hurts perf